compiler/src/dmd/e2ir.d

```diff
@@ -5757,7 +5757,25 @@ bool elemIsLvalue(elem* e)
         return elemIsLvalue(ec.EV.E1) && elemIsLvalue(ec.EV.E2);
     }
 
-    return e.Eoper == OPvar || e.Eoper == OPind;
+    if (e.Eoper == OPvar)
+        return true;
+
+    /* Match *(&__tmpfordtor+0) which is being destroyed
+     */
+    elem* ev;
+    if (e.Eoper == OPind &&
+        e.EV.E1.Eoper == OPadd &&
+        e.EV.E1.EV.E2.Eoper == OPconst &&
+        e.EV.E1.EV.E1.Eoper == OPaddr &&
+        (ev = e.EV.E1.EV.E1.EV.E1).Eoper == OPvar)
+    {
+        if (strncmp(ev.EV.Vsym.Sident.ptr, "__tmpfordtor", 12) == 0)
```

Comment:
There should be a flag that identifies this variable. Not a name comparison.

Reply:
It's a bit problematic, as the variable is created in the front end, and here is referenced in the back end. Adding a flag would add dependency to 3 additional files, including code to transfer it from the front end to the back. It isn't worth it.

Reply:
Isn't there a storage class for destructoring variables?

Reply:
And why is it you can't compose the backend IR as:
```
try
{
    __tmpfordtor = bar();
    try
    {
        _etmp = val(&__tmpfordtor);
        _retval = _*etmp;
    }
    finally
    {
        __dtor(&__tmpfordtor);
    }
}
finally
{
}
return _retval;
```

This is how gdc (and I guess ldc as well) avoid hitting the codegen bug.

Reply:
That's how dmd works now, too.

Reply:
Doesn't seem like it if you're resorting to matching local variable names.  I'll await the regression report when someone notices a behaviour change when they rename a variable from `_tmpfordtor` to `__tmpfordtor`. ;-)